### PR TITLE
close issue #93 accept any string format for tense input

### PR DESCRIPTION
**Pull request by yjlim5**
- change the tense strings to be lower case
- accept input even if it's capitalized
- add test fro `'should conjugate whether tense input is capitalized or all lowercase'`

### CODE REVIEW

File: src/french.js [hunk 1]
```diff
@@ -94,62 +94,62 @@ class French {
...
+      case 'subjonctif':
+        switch (info.tense.toLowerCase()) {
+          case 'présent':
+          case 'passé':
+          case 'imparfait':
+          case 'plus-que-parfait':
             return 'not implemented yet';
           default:
-            return 'Subjonctif Tense not found';
+            return 'subjonctif Tense not found';
```
Comment (cecilebertin) [inline src/french.js]: Lower case in message here not necessary

File: src/hindi.js [hunk 2]
```diff
@@ -10,27 +10,27 @@ class Hindi {
...
-    // {tense: 'Past/Present/Future', gender: 'male/female'}
+    // {tense: 'past/present/future', gender: 'male/female'}
     word = word.trim().toLowerCase();
 
-    switch (info.tense) {
-      case 'Present':
-        if (info.gender === 'male') {
+    switch (info.tense.toLowerCase()) {
+      case 'present':
+        if (info.gender && info.gender.toLowerCase() === 'male') {
```
Comment (cecilebertin) [inline src/hindi.js:18]: Why && info.gender.toLowerCase() and not just:
info.gender.toLowerCase() === 'male' ?

File: src/hindi.js [hunk 3]
```diff
@@ -10,27 +10,27 @@ class Hindi {
...
           return word.replace(/na$/, 'ta hun');
         }
         return word.replace(/na$/, 'ti hun');
-      case 'Past':
-        if (info.gender === 'male') {
+      case 'past':
+        if (info.gender && info.gender.toLowerCase() === 'male') {
```
Comment (cecilebertin) [inline src/hindi.js:23]: Why && info.gender.toLowerCase() and not just:
info.gender.toLowerCase() === 'male' ?

File: src/hindi.js [hunk 4]
```diff
@@ -10,27 +10,27 @@ class Hindi {
...
           return word.replace(/na$/, 'ta tha');
         }
         return word.replace(/na$/, 'ti thi');
-      case 'Future':
-        if (info.gender === 'male') {
+      case 'future':
+        if (info.gender && info.gender.toLowerCase() === 'male') {
```
Comment (cecilebertin) [inline src/hindi.js:28]: Why && info.gender.toLowerCase() and not just:
info.gender.toLowerCase() === 'male' ?

File: src/korean.js [hunk 5]
```diff
@@ -46,27 +46,27 @@ class Korean {
...
         const futureConjugation = this.doFuture(word);
         return futureConjugation.substring(0, futureConjugation.length - 3);
       }
       default:
-        return 'Could not find any rules for conjugation';
+        return `Could not find any rules for ${info.tense}`;
     }
 
-    if (info.formality === 'formal') {
+    if (info.formality && info.formality.toLowerCase() === 'formal') {
```
Comment (cecilebertin) [inline src/korean.js:69]: Why && info.formality.toLowerCase() and not just:
info.formality.toLowerCase() === 'male' ?

**Review by cecilebertin — CHANGES_REQUESTED** [review]

File: src/korean.js (hunk 5 above)
Comment (yjlim5) [reply src/korean.js:69]: During testing, I found out that some inputs may not have formality specified and only have tense.  It throws error if we try to call toLowerCase() on an undefined value.  Same for hindi files.

File: src/french.js (hunk 1 above)
Comment (yjlim5) [reply src/french.js]: done.

**Review by cecilebertin — APPROVED** [review]

**Pull request merged**